src/store/reset-password/saga.ts

```diff
@@ -0,0 +1,52 @@
+import { resetPassword as resetPasswordApi } from './api';
+import { setLoading, setErrors, SagaActionTypes, setEmailSubmitted } from './index';
+import { ResetPasswordErrors } from '.';
+import { call, put, take } from 'redux-saga/effects';
+
+export function validateResetPassword({ email }) {
+  const validationErrors = [];
+
+  if (!email.trim()) {
+    validationErrors.push(ResetPasswordErrors.EMAIL_REQUIRED);
+  }
+
+  return validationErrors;
+}
+
+export function* resetPasswordPage() {
+  let success;
+  do {
+    const action = yield take(SagaActionTypes.ResetPassword);
+    yield put(setLoading(true));
+    yield put(setEmailSubmitted(false));
+
+    const validationErrors = validateResetPassword(action.payload);
+
+    if (validationErrors.length > 0) {
+      yield put(setErrors(validationErrors));
+      success = false;
+      continue;
+    }
+
+    const result = yield call(resetPasswordApi, action.payload);
+    success = result.success;
+
+    if (success) {
+      yield put(setEmailSubmitted(true));
+    } else {
+      let errors = [];
+      if (result.response === 'EMAIL_NOT_FOUND') {
+        errors.push(ResetPasswordErrors.EMAIL_NOT_FOUND);
+      } else {
+        errors.push(ResetPasswordErrors.UNKNOWN_ERROR);
+      }
+      yield put(setErrors(errors));
+    }
+
+    yield put(setLoading(false));
+  } while (!success);
+}
+
+export function* saga() {
+  yield resetPasswordPage();
```

Comment:
I'm not sure this pattern is quite right. This seems like you'd only be able to reach the reset password stage a single time. What happens if I request a password reset and then click the `login` link on the footer and then click the link to try to come back and reset it again? My guess is the user would be stuck.

We also don't really want to be waiting for password reset events if you're actually logged into the app. It's cleaner to only be listening for these events when we expect them to happen (i.e., when we've gotten to that page)